src/init.cpp

```diff
@@ -1702,6 +1720,11 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         }
     }
 
+    if (fTrimHeaders) {
+        LogPrintf("Unsetting NODE_NETWORK_LIMITED on header trim mode\n");
+        nLocalServices = ServiceFlags(nLocalServices & ~NODE_NETWORK_LIMITED);
```

Comment:
This `NODE_NETWORK_LIMITED` flag already existed in the code base, right? And it means that the node will provide peers with some blocks, but not all?

Am I correct in assuming that that unsetting it indicates that the node will not provide any blocks to peers?

I'm curious as to why this already existed... is there an existing setting in the conf file which enables you to launch a node with this flag set? Or why did it already exist?

Reply:
It exists for SPV / "light" nodes. I think Bitcoin Core nodes would never set it, they would only read it.